tests/run/duplicate_utilitycode.pyx

```diff
@@ -0,0 +1,27 @@
+# mode: run
+
+# There were a few cases where duplicate utility code definitions (i.e. with the same name)
+# could be generated, causing C compile errors. This file tests them.
```

Comment:
These tests can go into `tests/run/cfunc_convert.pyx` (took me a while to find it – this feature seems to lack an obvious name). I don't think we need a separate test file for this specific issue. It's really related to the feature itself.

Note that you can add the ticket number as tag: `# ticket: gh3716`

And let's also see if both wrapper functions correctly accept the expected keywords:
```python
    >>> print(make_map()['f1'](x=1, r=2))
    f1
    >>> make_map()['f1'](x1=1, r=2)  # doctest: +ELLIPSIS
    Traceback (most recent call last):
    TypeError: ...
    >>> print(make_map()['f2'](x1=1, r=2))
    f2
    >>> make_map()['f2'](x=1, r=2)  # doctest: +ELLIPSIS
    Traceback (most recent call last):
    TypeError: ...
```

Reply:
Moved and added.